src/main/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStep.java

```diff
@@ -118,22 +171,42 @@ public String getDisplayName() {
         @Override
         public void load() {
             super.load();
-            if (milestonesByOrdinalByJob == null) {
-                milestonesByOrdinalByJob = new TreeMap<String, Map<Integer, Milestone>>();
+            if (milestonesByGroupByJob == null) {
+                milestonesByGroupByJob = new TreeMap<String, Map<String, Milestone>>();
             }
-            LOGGER.log(Level.FINE, "load: {0}", milestonesByOrdinalByJob);
+            else {
+                this.milestonesByGroupByJob.forEach((j,m) -> m.forEach( (o,v) -> LOGGER.log(Level.INFO, "loaded milestone: {0}", o)));
+            }
+            LOGGER.log(Level.INFO, "load: {0}", milestonesByGroupByJob);
+        }
+
+        public Object readResolve() {
+            LOGGER.log(Level.INFO, "readResolve: {0}", this);
+            LOGGER.log(Level.INFO, "readResolve: this.milestonesByGroupByJob: {0}", this.milestonesByGroupByJob);
+            LOGGER.log(Level.INFO, "readResolve: this.milestonesByOrdinalByJob: {0}", this.milestonesByOrdinalByJob);
```

Comment:
@jglick, 

I've added test scenario and pre-defined configuration data. I hoped that if I restore milestonesByOrdinalByJob field I'll be able to load old config`migration_1_3to1_4/org.jenkinsci.plugins.pipeline.milestone.MilestoneStep.xml`

unfortunately I got `null`s
```
 2.331 [id=28]	INFO	o.j.p.p.m.MilestoneStep$DescriptorImpl#readResolve: readResolve: org.jenkinsci.plugins.pipeline.milestone.MilestoneStep$DescriptorImpl@417b0e3
   2.331 [id=28]	INFO	o.j.p.p.m.MilestoneStep$DescriptorImpl#readResolve: readResolve: this.milestonesByGroupByJob: null
   2.332 [id=28]	INFO	o.j.p.p.m.MilestoneStep$DescriptorImpl#readResolve: readResolve: this.milestonesByOrdinalByJob: null
   2.332 [id=28]	INFO	o.j.p.p.m.MilestoneStep$DescriptorImpl#load: load: {}
   3.039 [id=26]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
```

Reply:
Not sure, but anyway if you have a `load` method I suppose there is no need for `readResolve`.